server/schedulers/mockcluster.go

```diff
@@ -0,0 +1,529 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schedulers
+
+import (
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/juju/errors"
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/kvproto/pkg/pdpb"
+	"github.com/pingcap/pd/server/cache"
+	"github.com/pingcap/pd/server/core"
+)
+
+type mockCluster struct {
+	id              *core.MockIDAllocator
+	stores          *core.StoresInfo
+	regions         *core.RegionsInfo
+	writeStatistics cache.Cache
+	readStatistics  cache.Cache
+}
+
+// NewMockCluster creates a new mockCluster
+func newMockCluster(id *core.MockIDAllocator) *mockCluster {
+	return &mockCluster{
```

Comment:
maybe we can let `clusterinfo` be public and use it directly

Reply:
ok， I'll correct it.